src/sugar.rs

```diff
@@ -674,6 +673,25 @@ macro_rules! proptest_helper {
     (@_WRAPSTR ($a:pat, $($rest:pat),*)) => {
         (stringify!($a), proptest_helper!(@_WRAPSTR ($($rest),*)))
     };
+    // build a property testing block that when executed, executes the full property test.
+    (@_BODY $config:ident ($($parm:pat in $strategy:expr),+) $body:block) => {{
```

Comment:
The code that sets the source file should probably be here so that it also applies to the new case.

Reply:
Done. I kept the `.clone()` in the calling macro to match how it behaved with `clone_with_source_file()`.